core-bundle/src/Resources/contao/classes/BackendUser.php

```diff
@@ -601,28 +601,48 @@ public function getRoles()
 	}
 
 	/**
-	 * {@inheritdoc}
+	 * @deprecated Deprecated since Contao 4.12, to be removed in Contao 5.0.
 	 */
 	public function serialize()
 	{
-		return serialize(array('admin' => $this->admin, 'amg' => $this->amg, 'parent' => parent::serialize()));
+		$data = $this->__serialize();
+		$data['parent'] = serialize($data['parent']);
+
+		return serialize($data);
+	}
+
+	public function __serialize(): array
+	{
+		return array('admin' => $this->admin, 'amg' => $this->amg, 'parent' => parent::__serialize());
 	}
 
 	/**
-	 * {@inheritdoc}
+	 * @deprecated Deprecated since Contao 4.12, to be removed in Contao 5.0.
 	 */
 	public function unserialize($serialized)
 	{
 		$data = unserialize($serialized, array('allowed_classes'=>false));
 
+		if (!isset($data['parent']))
+		{
+			return;
```

Comment:
Is this correct? Because it prevents `$this->__unserialize($data);` from being called if `$data['parent']` is not set.

Reply:
IMO yes. If `$data['parent']` is not set, the data is in an unexpected format and it cannot be unserialized.

Reply:
It worked the same way before: 

https://github.com/contao/contao/blob/43188094917d7e29242c7bf49418331a2f74f851/core-bundle/src/Resources/contao/classes/BackendUser.php#L611-L614

`parent` was required to be set.